x-pack/plugins/cases/server/client/attachments/bulk_get.ts

```diff
@@ -0,0 +1,177 @@
+    logger,
+    authorization,
+  } = clientArgs;
+
+  try {
+    const request = pipe(
+      excess(BulkGetAttachmentsRequestRt).decode({ ids: attachmentIDs }),
+      fold(throwErrors(Boom.badRequest), identity)
+    );
+
+    throwErrorIfIdsExceedTheLimit(request.ids);
+
+    const attachments = await attachmentService.getter.bulkGet(request.ids);
+
+    const { validAttachments, attachmentsWithErrors, invalidAssociationAttachments } =
+      partitionAttachments(caseID, attachments);
+
+    const { authorized: authorizedAttachments, unauthorized: unauthorizedAttachments } =
+      await authorization.getAndEnsureAuthorizedEntities({
+        savedObjects: validAttachments,
+        operation: Operations.bulkGetAttachments,
+      });
+
+    const errors = constructErrors({
+      associationErrors: invalidAssociationAttachments,
+      unauthorizedAttachments,
+      soBulkGetErrors: attachmentsWithErrors,
+      caseId: caseID,
+    });
+
+    return BulkGetAttachmentsResponseRt.encode({
+      attachments: flattenCommentSavedObjects(authorizedAttachments),
+      errors,
+    });
+  } catch (error) {
+    throw createCaseError({
+      message: `Failed to bulk get attachments for case id: ${caseID}: ${error}`,
+      error,
+      logger,
+    });
+  }
+}
+
+const throwErrorIfIdsExceedTheLimit = (ids: string[]) => {
+  if (ids.length > MAX_BULK_GET_ATTACHMENTS) {
+    throw Boom.badRequest(
+      `Maximum request limit of ${MAX_BULK_GET_ATTACHMENTS} attachments reached`
+    );
+  }
+};
+
+interface PartitionedAttachments {
+  validAttachments: AttachmentSavedObject[];
+  attachmentsWithErrors: AttachmentSavedObjectWithErrors;
+  invalidAssociationAttachments: AttachmentSavedObject[];
+}
+
+const partitionAttachments = (
+  caseId: string,
+  attachments: BulkOptionalAttributes<CommentAttributes>
+): PartitionedAttachments => {
+  const [attachmentsWithoutErrors, errors] = partitionBySOError(attachments.saved_objects);
+  const [caseAttachments, invalidAssociationAttachments] = partitionByCaseAssociation(
+    caseId,
+    attachmentsWithoutErrors
+  );
+
+  return {
+    validAttachments: caseAttachments,
+    attachmentsWithErrors: errors,
+    invalidAssociationAttachments,
+  };
+};
+
+const partitionBySOError = (attachments: Array<OptionalAttributes<CommentAttributes>>) =>
+  partition(
+    attachments,
+    (attachment) => attachment.error == null && attachment.attributes != null
+  ) as [AttachmentSavedObject[], AttachmentSavedObjectWithErrors];
+
+const partitionByCaseAssociation = (caseId: string, attachments: AttachmentSavedObject[]) =>
+  partition(attachments, (attachment) => {
+    const ref = getCaseReference(attachment.references);
+
+    return caseId === ref?.id;
+  });
+
+const getCaseReference = (references: SavedObjectReference[]): SavedObjectReference | undefined => {
+  return references.find(
+    (ref) => ref.name === `associated-${CASE_SAVED_OBJECT}` && ref.type === CASE_SAVED_OBJECT
```

Comment:
I think it is better if we use a constant for the reference name as we do here `x-pack/plugins/cases/server/common/constants.ts`.